streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java

```diff
@@ -265,7 +265,7 @@ private void verifyAllKVKeys(final StreamRunnable[] streamRunnables, final Kafka
                 public boolean conditionMet() {
                     try {
                         final StreamsMetadata metadata = streams.metadataForKey(storeName, key, new StringSerializer());
-                        if (metadata == null) {
+                        if (metadata == null || metadata.equals(StreamsMetadata.NOT_AVAILABLE)) {
```

Comment:
People might call it with an invalid `storeName`, so it will never exist. I think it should throw in this case. @guozhangwang - thoughts?

Yep, we just had a long conversation about this on another PR.  Stream re-init is not really exceptional and people should back-off and re-try in these cases. Though, i could argue for throwing an exception here, too :-)